lib/writers/git.go

```diff
@@ -95,10 +102,40 @@ func NewGitWriter(logger logr.Logger, stateStoreSpec v1alpha1.GitStateStoreSpec,
 			return nil, fmt.Errorf("password not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
 		}
 
-		authMethod = &http.BasicAuth{
+		authMethod = &githttp.BasicAuth{
 			Username: string(username),
 			Password: string(password),
 		}
+	case v1alpha1.GitHubAppAuthMethod:
+		appID, ok := creds["appID"]
+		if !ok {
+			return nil, fmt.Errorf("appID not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
+		}
+		installationID, ok := creds["installationID"]
+		if !ok {
+			return nil, fmt.Errorf("installationID not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
+		}
+		privateKey, ok := creds["privateKey"]
+		if !ok {
+			return nil, fmt.Errorf("privateKey not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
+		}
+
+		j, err := GenerateGitHubAppJWT(string(appID), string(privateKey))
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate GitHub App JWT: %w", err)
+		}
+
+		token, exp, err := GetGitHubInstallationTokenWithExpiry(string(installationID), j)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get GitHub installation token: %w", err)
+		}
+
+		ba := &githttp.BasicAuth{
+			Username: "x-access-token",
+			Password: token,
+		}
+		authMethod = ba
+		StartGitHubTokenAutoRefresh(logger, ba, string(appID), string(installationID), string(privateKey), exp)
```

Comment:
question: the GitWriter is not a singleton object, we create a brand new one every time we reconcile (see, for example, the [Work Controller](https://github.com/syntasso/kratix/blob/ce69df0831889d52294346163a7e4112b4c3b85c/internal/controller/workplacement_controller.go#L334)). With my reading of the code, it is:

* Generating a new token every time we create the object
* Starting a new Go Routine to refresh the token

So question 1:
* Do we need to set up the token refresh routine?
* If we do need the refresh, should we then provide a way to cancel the previous routine? Otherwise we will just keep creating them until Kratix gets killed

Am I missing something?